src/main/java/org/folio/service/inventory/InventoryManager.java

```diff
@@ -1059,12 +1067,17 @@ public Future<List<JsonObject>> getItemsByPoLineIdsAndStatus(List<String> poLine
   }
 
   public Future<CompositePoLine> openOrderHandleInstance(CompositePoLine compPOL, boolean isInstanceMatchingDisabled, RequestContext requestContext) {
-    if (compPOL.getInstanceId() != null) {
-      return Future.succeededFuture(compPOL);
-    } else {
-      return getInstanceRecord(compPOL, isInstanceMatchingDisabled, requestContext)
-        .map(compPOL::withInstanceId);
-    }
+    return consortiumConfigurationService.getConsortiumConfiguration(requestContext)
+      .compose(consortiumConfiguration -> {
+        if (compPOL.getInstanceId() != null) {
+          return consortiumConfiguration.isEmpty() ? Future.succeededFuture(compPOL.getInstanceId())
+            : shareInstanceAmongTenantsIfNeeded(compPOL.getInstanceId(), consortiumConfiguration.get(), compPOL.getLocations(), requestContext);
+        } else {
+          return getInstanceRecord(compPOL, isInstanceMatchingDisabled, requestContext)
+            .compose(instanceId -> consortiumConfiguration.isEmpty() ? Future.succeededFuture(instanceId)
+              : shareInstanceAmongTenantsIfNeeded(instanceId, consortiumConfiguration.get(), compPOL.getLocations(), requestContext));
+        }
```

Comment:
maybe action is not required, just idea about removing else, you will decide
https://developer20.com/elseless/#:~:text=One%20of%20the%20main%20arguments,control%20flow%20of%20a%20program.

Reply:
in case of early returns without any business logic I can remove else but it's definetelly another one